app/assets/javascripts/discourse/components/backup-codes.js.es6

```diff
@@ -0,0 +1,58 @@
+import computed from "ember-addons/ember-computed-decorators";
+
+export default Ember.Component.extend({
+  classNames: ["backup-codes"],
+  backupCodes: null,
+
+  click(event) {
+    if (event.target.id === "backupCodes") {
+      this._selectAllBackupCodes();
+    }
+  },
+
+  didRender() {
+    this._super();
+
+    const $backupCodes = this.$("#backupCodes");
+    if ($backupCodes.length) {
+      $backupCodes.height($backupCodes[0].scrollHeight);
+    }
+  },
+
+  @computed("formattedBackupCodes")
+  base64BackupCode(formattedBackupCodes) {
+    return this._b64EncodeUnicode(formattedBackupCodes);
+  },
+
+  @computed("backupCodes")
+  formattedBackupCodes(backupCodes) {
+    if (!backupCodes) return null;
+
+    return backupCodes.join("\n").trim();
+  },
+
+  actions: {
+    copyToClipboard() {
+      this._selectAllBackupCodes();
+      this.get("copyBackupCode")(document.execCommand("copy"));
+    }
+  },
+
+  // https://developer.mozilla.org/en-US/docs/Web/API/WindowBase64/Base64_encoding_and_decoding
+  _b64EncodeUnicode(str) {
```

Comment:
This is a small thing, but I noticed `_b64EncodeUnicode` doesn't use `this` in any way, so it's a pure function. For such functions, I often prefer to put them outside of the class as a regular function. Funnily enough, it is more "private" because it is not exported from your module, and can only be called by the class (whereas any code could call `_b64EncodeUnicode` if they wanted).

Once you have a pure function, you can also shorten the `base64BackupCode` computed property by a couple of lines up on L23:

```
@computed("formattedBackupCodes")
base64BackupCode: b64EncodeUnicode
```